### PR TITLE
fix(radio): stops onSelect firing twice every time we click a radio

### LINKED ISSUE #657: radio: onSelect firing twice every time we click a radio button
image (4)
From Travis Bumgarner
onSelect is called both when an item is deselected and when a new one is selected.

### PR DESCRIPTION
**Pull request by EdwardIrby**
fix #657

### CODE REVIEW

**Comment by EdwardIrby** [comment]
> Were you able to repro this? It'd be nice to have a failing test that this PR would then address. What do you think?

I was able to reproduce this. I'll write a test to show it's not happening.

File: packages/radio/src/react/__specs__/index.spec.js [hunk 1]
```diff
@@ -19,4 +19,19 @@ describe('Radio', () => {
     expect(group.current).not.toBeNull()
     expect(button.current).not.toBeNull()
   })
+  it('fires event once', () => {
+    const group = React.createRef()
```
Comment (danethurber) [inline packages/radio/src/react/__specs__/index.spec.js]: these refs aren't needed for this test

File: packages/radio/src/react/__specs__/index.spec.js [hunk 2]
```diff
@@ -19,4 +19,19 @@ describe('Radio', () => {
...
+    const button = React.createRef()
+    const spy = jest.fn()
+    const { container } = render(
+      <Radio.Group ref={group} onSelect={spy}>
+        <Radio.Button value="red" label="Red" />
+        <Radio.Button ref={button} value="green" label="Green" />
+        <Radio.Button value="blue" label="Blue" />
+      </Radio.Group>
+    )
+    const rdio = container.querySelector('label')
```
Comment (danethurber) [inline packages/radio/src/react/__specs__/index.spec.js]: typo. should be `radio`

File: packages/radio/src/react/__specs__/index.spec.js (hunk 1 above)
Comment (EdwardIrby) [reply packages/radio/src/react/__specs__/index.spec.js]: fixed

File: packages/radio/src/react/__specs__/index.spec.js (hunk 2 above)
Comment (EdwardIrby) [reply packages/radio/src/react/__specs__/index.spec.js]: fixed

**Pull request merged**